### ISSUE TITLE
no-implicit-conversion should not warn on "" + "foo"

### ISSUE THREAD

**Issue by TheSavior** [eslint/eslint#7057]
**What version of ESLint are you using?**
3.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
Running the test suite

**What did you do? Please include the actual source code causing the issue.**
I added a test that the following should be valid:

```
{code: "1 * 1"},
{code: "'' + 'foo'"},
```

and ran 

```
./node_modules/.bin/mocha /Users/eli/projects/eslint/tests/lib/rules/no-implicit-coercion.js
```

**What did you expect to happen?**
I expected these to be valid

**What actually happened? Please include the actual, raw output from ESLint.**
eslint was fine with `1 * 1` but not with `"" + "foo"`. 

While both of these aren't very good patterns for people to be following, there is nothing happening with type conversion here and thus `no-implicit-conversion` should not care about these cases.

**Comment by platinumazure** [comment]
Just to be clear, the issue here is that there is no conversion because the other operand is definitely the same type?

**Comment by kaicataldo** [comment]
Seems like a bug to me, give that no coercion is happening.  Confirmed locally.

**Comment by TheSavior** [comment]
@platinumazure yep, exactly.
